packages/datepicker-react/src/DatePicker.tsx

```diff
@@ -74,7 +99,63 @@ export function DatePicker({
                     </div>
                     <table data-testid="jkl-datepicker-calendar" />
                 </CoreDatepicker>
-            </CoreToggle>
+            </div>
+            <SupportLabel errorLabel={errorLabel} helpLabel={helpLabel} />
         </div>
     );
 }
+
+function useOnClickOutside<T extends HTMLElement>(onClickOutside: () => void): MutableRefObject<T | null> {
+    const ref = useRef<T>(null);
+
+    function onMouseDown(e: MouseEvent | FocusEvent) {
+        if (ref.current && !ref.current.contains(e.target as Node)) {
+            onClickOutside();
+        }
+    }
+
+    useEffect(() => {
+        // Bind the event listener
+        document.addEventListener("mousedown", onMouseDown);
+        // @ts-ignore
+        document.addEventListener("focusin", onMouseDown);
+
+        return () => {
+            // Unbind the event listener on clean up
+            document.removeEventListener("mousedown", onMouseDown);
+            // @ts-ignore
+            document.removeEventListener("focusin", onMouseDown);
+        };
+    }, []);
+    return ref;
+}
+
+function isSameDay(date1: Date, date2: Date) {
+    return (
+        date1.getDate() === date2.getDate() &&
+        date1.getMonth() === date2.getMonth() &&
+        date1.getFullYear() === date2.getFullYear()
+    );
+}
+
+/**
+ *
+ * @param date the date to format
+ * @return returns a date with "dd.mm.yyyy"-format
+ */
+function formatDate(date: Date) {
+    const day = pad0(date.getDate());
```

Comment:
`date.getDate().padStart(2,"0")` ES7 FTW!

Reply:
Not quite the same, though. Unlike the current function it will pad 0 to 00, but I guess that should never be tha case anyway 🤷‍♂️

Reply:
Also, I don't think padStart accepts numbers?

Reply:
`0` could never happen, if so, shit has hit the fan. But yeah, number is not valid, ``${date.getDate()}`.pad(2,"0")` should be fine.

You decide, but I would prefer a unit test for pad0(if we keep it) and formatDate.